### PR TITLE
Fix amp-sidebar scrolling on open/close in UIWebView

### PR DESCRIPTION
**Pull request by choumx**
`HTMLElement.focus()` causes undesirable scrolling on UIWebView. There also doesn't seem to be a good way to target UIWebView specifically via user agent.

Tested on iPhone 7, Safari 12.1.3 and this doesn't appear to have any impact on a11y (VoiceOver focus).

`b/133327438`

### CODE REVIEW

File: extensions/amp-sidebar/0.1/amp-sidebar.js [hunk 1]
```diff
@@ -368,7 +371,10 @@ export class AmpSidebar extends AMP.BaseElement {
       this.historyId_ = -1;
     }
     if (this.openerElement_ && sidebarIsActive && scrollDidNotChange) {
-      tryFocus(this.openerElement_);
+      // As of iOS 12.2, focus() causes undesired scrolling in UIWebViews.
```
Comment (aghassemi) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: How about restricting to `const iOSWebview = viewer.isWebviewEmbedded() && platform.isIos();` instead?

(this `isWebviewEmbedded` returns based on whether viewer is setting `webview` param on the URL, if the partner here is not setting that, they probably should, there are a few other places that checks this condition)

Comment (choumx) [reply extensions/amp-sidebar/0.1/amp-sidebar.js]: Good idea, done.

Comment (choumx) [reply extensions/amp-sidebar/0.1/amp-sidebar.js]: Actually, `isWebviewEmbedded()` assumes there's no parent iframe inside the web view. This isn't necessarily true, so I changed it to `isEmbedded()`.

**Comment by sparhami** [comment]
@choumx This breaks accessibility for restoring focus to the opener element when closing. This doesn't break accessibility for opening the sidebar because that was broken to begin with.

We're going to need to undo this change. Did you explore any alternative approaches for this? If so, what alternatives did you explore and what problems did you encounter?

**Comment by choumx** [comment]
Referenced bug has more details. Have there been user complaints? 

We can try to use a more targeted fix though detecting UIWebView seems difficult.

**Pull request merged**